deployer/src/persistence.rs

```diff
@@ -59,14 +62,27 @@ impl Persistence {
         let (log_send, mut log_recv): (UnboundedSender<deploy_layer::Log>, _) =
             mpsc::unbounded_channel();
 
+        let (build_log_send, _) = broadcast::channel(32);
+        let build_log_send_clone = build_log_send.clone();
+
         let pool_cloned = pool.clone();
 
         // The logs are received on a non-async thread.
         // This moves them to an async thread
         let handle = tokio::spawn(async move {
             while let Some(log) = log_recv.recv().await {
                 match log.r#type {
-                    LogType::Event => insert_log(&pool_cloned, log).await.unwrap(),
+                    LogType::Event => {
+                        if log.state == State::Building {
+                            if let Some(build_log) = Into::into(&log) {
+                                build_log_send_clone
+                                    .send(build_log)
+                                    .expect("failed to broadcast build log");
```

Comment:
In this, a thread is spawned which single handedly manages the flow of of `Log`s to places where it needs to go. If one of these `.expect` or `.unwrap` actually panics, I guess the service is bricked but it will keep running instead of restarting, or fixing the condition.

We should perhaps avoid things that could panic in here (and maybe log them out as errors instead), and later on grab this returned `handle` and monitor it?